File: geosave/app/src/main/java/com/jakdor/geosave/common/model/ElevationApi.java

package com.jakdor.geosave.common.model;

import java.io.Serializable;
import java.util.List;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class ElevationApi implements Serializable {

    @SerializedName("elevationApiResults")
    @Expose
    private List<ElevationApiResult> elevationApiResults = null;
    private final static long serialVersionUID = 8255691863523063062L;

    /**
     * No args constructor for use in serialization
     */
    public ElevationApi() {
    }

    public ElevationApi(List<ElevationApiResult> elevationApiResults) {
        super();
        this.elevationApiResults = elevationApiResults;
    }

    public List<ElevationApiResult> getElevationApiResults() {
        return elevationApiResults;
    }

    public void setElevationApiResults(List<ElevationApiResult> elevationApiResults) {
        this.elevationApiResults = elevationApiResults;
    }
}

File: geosave/app/src/main/java/com/jakdor/geosave/common/model/ElevationApi.kt
/*
 * GeoSave - app for easy sharing and collaborating on GPS related data
 * Copyright (C) 2018  Jakub Dorda
 *
 * Software under GPLv3 licence - full copyright notice available at:
 * https://github.com/jakdor/geosave/blob/master/README.md
 */

package com.jakdor.geosave.common.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class ElevationApi
constructor(@SerializedName("results")
            @Expose var elevationApiResults: List<ElevationApiResult>)
